common/djangoapps/track/shim.py

```diff
@@ -122,13 +127,38 @@ def __call__(self, event):
         if 'current_time' in payload:
             payload['currentTime'] = payload.pop('current_time')
 
-        event['event'] = json.dumps(payload)
-
-        if 'context' not in event:
-            return
-
-        context = event['context']
+        if 'context' in event:
+            context = event['context']
+
+            # For converting seek_type to seek and skip|slide to onSlideSeek|onSkipSeek
+            if 'seek_type' in payload:
+                seek_type = payload['seek_type']
+                if seek_type == 'slide':
+                    payload['type'] = "onSlideSeek"
+                elif seek_type == 'skip':
+                    payload['type'] = "onSkipSeek"
+                del payload['seek_type']
+
+            # For the iOS build that is returning a +30 for back skip 30
+            if (
+                context['application']['version'] == "1.0.02" and
+                context['application']['name'] == "edx.mobileapp.iOS"
+            ):
+                if 'requested_skip_interval' in payload and 'type' in payload:
+                    if (
+                        payload['requested_skip_interval'] == 30 and
+                        payload['type'] == "onSkipSeek"
+                    ):
+                        payload['requested_skip_interval'] = -30
+
+            #  For the Android build that isn't distinguishing between skip and seek
+            if 'requested_skip_interval' in payload:
+                if abs(payload['requested_skip_interval']) != 30:
+                    if 'type' in payload:
+                        payload['type'] = 'onSlideSeek'
+
+            if 'open_in_browser_url' in context:
+                page, _sep, _tail = context.pop('open_in_browser_url').rpartition('/')
+                event['page'] = page
 
-        if 'open_in_browser_url' in context:
-            page, _sep, _tail = context.pop('open_in_browser_url').rpartition('/')
-            event['page'] = page
```

Comment:
So before this PR, the 'page' field modification here was not included as part the `json.dumps`.  I hope this won't cause any issues.  I wouldn't think so - but what do I know about what our data packages need?